website/content/docs/v0.14/Guides/storage.md

```diff
@@ -78,112 +80,33 @@ To see a live demo of this section, see the video below:
+We need to set the `vm.nr_hugepages` sysctl and add `openebs.io/engine=mayastor` labels to the nodes which are meant to be storage nodes
+This can be done with `talosctl patch machineconfig` or via config patches during `talosctl gen config`.
 
-- Either using `kubectl taint node` in a pre-existing cluster or by updating `/machine/kubelet/extraArgs` in machine config, add `openebs.io/engine=mayastor` as a node label.
-If being done via machine config, `extraArgs` may look like:
+Some examples are shown below, modify as needed.
 
-```yaml
-...
-extraArgs:
-  node-labels: openebs.io/engine=mayastor
-...
-```
-
-### Deploy Mayastor
-
-Using the [Mayastor docs](https://mayastor.gitbook.io/introduction/quickstart/deploy-mayastor) as a reference, apply all YAML files necessary.
-At the time of writing this looked like:
+Using gen config
 
 ```bash
-kubectl create namespace mayastor
-
-kubectl apply -f https://raw.githubusercontent.com/openebs/Mayastor/master/deploy/moac-rbac.yaml
-
-kubectl apply -f https://raw.githubusercontent.com/openebs/Mayastor/master/deploy/nats-deployment.yaml
-
-kubectl apply -f https://raw.githubusercontent.com/openebs/Mayastor/master/csi/moac/crds/mayastorpool.yaml
-
-kubectl apply -f https://raw.githubusercontent.com/openebs/Mayastor/master/deploy/csi-daemonset.yaml
-
-kubectl apply -f https://raw.githubusercontent.com/openebs/Mayastor/master/deploy/moac-deployment.yaml
-
-kubectl apply -f https://raw.githubusercontent.com/openebs/Mayastor/master/deploy/mayastor-daemonset.yaml
+talosctl gen config my-cluster https://mycluster.local:6443 --config-patch '[{"op": "add", "path": "/machine/sysctls", "value": {"vm.nr_hugepages": "1024"}}, {"op": "add", "path": "/machine/kubelet/extraArgs", "value": {"node-labels": "openebs.io/engine=mayastor"}}]'
 ```
 
-### Create Pools
-
-Each "storage" node should have a "MayastorPool" that defines the local disks to use for storage.
-These are later considered during scheduling and replication of data.
-Create the pool by issuing the following, updating as necessary:
+Patching an existing node
 
 ```bash
-cat <<EOF | kubectl create -f -
-apiVersion: "openebs.io/v1alpha1"
-kind: MayastorPool
-metadata:
-  name: pool-on-talos-xxx
-  namespace: mayastor
-spec:
-  node: talos-xxx
-  disks: ["/dev/sdx"]
-EOF
+talosctl patch machineconfig -n <node ip> --config-patch '[{"op": "add", "path": "/machine/sysctls", "value": {"vm.nr_hugepages": "1024"}}, {"op": "add", "path": "/machine/kubelet/extraArgs", "value": {"node-labels": "openebs.io/engine=mayastor"}}]'
 ```
 
-### Create StorageClass
-
-With the pools created for each node, create a storage class that uses the `nvmf` protocol, updating the number of replicas as necessary:
+> Note: If you are adding/updating the `vm.nr_hugepages` on a node which already had the `openebs.io/engine=mayastor` label set, you'd need to restart kubelet so that it picks up the new value, by issuing the following command
```

Comment:
I believe in 0.14 kubelet should be restarted if extraArgs are changed

Reply:
this is when the node already contain the labels and the sysctl has been changed, will it automatically restart kubelet? I had to restart kubelet manually for it to pick the new sysctl

Reply:
well, `sysctl` should be applied immediately, not sure if `kubelet` needs a restart to update something

I was referring more to the labels